src/ResourceManagement/EventHub/EventHubAuthorizationRulesImpl.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+namespace Microsoft.Azure.Management.Eventhub.Fluent
+{
+    using Microsoft.Azure.Management.EventHub.Fluent;
+    using Microsoft.Azure.Management.EventHub.Fluent.Models;
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Implementation for  EventHubAuthorizationRules.
+    /// </summary>
+    ///GENTHASH:Y29tLm1pY3Jvc29mdC5henVyZS5tYW5hZ2VtZW50LmV2ZW50aHViLmltcGxlbWVudGF0aW9uLkV2ZW50SHViQXV0aG9yaXphdGlvblJ1bGVzSW1wbA==
+    internal partial class EventHubAuthorizationRulesImpl  :
+        AuthorizationRulesBaseImpl<IEventHubsOperations,
+            Microsoft.Azure.Management.Eventhub.Fluent.IEventHubAuthorizationRule,
+            Microsoft.Azure.Management.Eventhub.Fluent.EventHubAuthorizationRuleImpl>,
+        IEventHubAuthorizationRules
+    {
+        ///GENMHASH:1642031856E30EE20D95D3D070FD2EB5:6591C3058C27F4799E76F6FECB0EC307
+        internal EventHubAuthorizationRulesImpl(IEventHubManager manager) : base(manager, manager.Inner.EventHubs)
+        {
+        }
+
+        ///GENMHASH:8ACFB0E23F5F24AD384313679B65F404:19B6B60B4510361D59C510F0C90DE4C9
+        public EventHubAuthorizationRuleImpl Define(string name)
+        {
+            return new EventHubAuthorizationRuleImpl(name, this.Manager);
+        }
+
+        ///GENMHASH:66A87AACDC6DD496194CABFCB3F2C921:E9A51CD479436338CBCF0CC9ED036074
+        public IEnumerable<Microsoft.Azure.Management.Eventhub.Fluent.IEventHubAuthorizationRule> ListByEventHub(string resourceGroupName, string namespaceName, string eventHubName)
+        {
+            return Extensions.Synchronize(() => this.Inner.ListAuthorizationRulesAsync(resourceGroupName,
+                    namespaceName,
+                    eventHubName))
+                .AsContinuousCollection(nextLink => Extensions.Synchronize(() => this.Inner.ListAuthorizationRulesNextAsync(nextLink))).Select(inner => WrapModel(inner));
+        }
+
+        ///GENMHASH:E776888E46F8A3FC56D24DF4A74E5B74:F081FE980E79F1E88B95F84B83248DFF
+        public override Task<Microsoft.Azure.Management.Eventhub.Fluent.IEventHubAuthorizationRule> GetByIdAsync(string id, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            ResourceId resourceId = ResourceId.FromString(id ?? throw new ArgumentNullException(id));
+            return GetByNameAsync(resourceId.ResourceGroupName,
```

Comment:
>            return GetByNameAsync(reso [](start = 0, length = 38)

async...await

Reply:
same comment as [this](https://github.com/Azure/azure-libraries-for-net/pull/179#discussion_r170814652)